python/lsst/ip/isr/isrStatistics.py

```diff
@@ -223,3 +291,157 @@ def measureCti(self, inputExp, overscans, gains):
             outputStats[amp.getName()] = ampStats
 
         return outputStats
+
+    @staticmethod
+    def makeKernel(kernelSize):
+        """Make a boxcar smoothing kernel.
+
+        Parameters
+        ----------
+        kernelSize : `int`
+            Length of the kernel in pixels.
+
+        Returns
+        -------
+        kernel : `np.array`
+            Kernel for boxcar smoothing.
+        """
+        if kernelSize > 0:
+            kernel = np.full(kernelSize, 1.0 / kernelSize)
+        else:
+            kernel = np.array([1.0])
+        return kernel
+
+    def measureBanding(self, inputExp, overscans):
+        """Task to measure banding statistics.
+
+        Parameters
+        ----------
+        inputExp : `lsst.afw.image.Exposure`
+            Exposure to measure.
+        overscans : `list` [`lsst.pipe.base.Struct`]
+            List of overscan results.  Expected fields are:
+
+            ``imageFit``
+                Value or fit subtracted from the amplifier image data
+                (scalar or `lsst.afw.image.Image`).
+            ``overscanFit``
+                Value or fit subtracted from the overscan image data
+                (scalar or `lsst.afw.image.Image`).
+            ``overscanImage``
+                Image of the overscan region with the overscan
+                correction applied (`lsst.afw.image.Image`). This
+                quantity is used to estimate the amplifier read noise
+                empirically.
+
+        Returns
+        -------
+        outputStats : `dict` [`str`, [`dict` [`str`,`float]]
+            Dictionary of measurements, keyed by amplifier name and
+            statistics segment.
+        """
+        outputStats = {}
+
+        detector = inputExp.getDetector()
+        kernel = self.makeKernel(self.config.bandingKernelSize)
+
+        outputStats['AMP_BANDING'] = []
+        for amp, overscanData in zip(detector.getAmplifiers(), overscans):
```

Comment:
I don't suppose the `overscanData` carries an amplifier id to check we're definitely zipping correctly?

Reply:
It does not, but it's generated by iterating over `detector.getAmplifiers()`, and has `None` added if overscan failed.  I think it will eventually evolve into a dict (with amp names as keys) in the future, because I've had similar concerns in the past.

Reply:
OK, sounds good. I think everywhere we can use ids we protect ourselves from future confusion/frustration/sadness.